### PR TITLE
Symlink for filenames

### PR DESCRIPTION
**Pull request by aayushKumarJarvis**
This PR is in reference to an issue which we experienced while integrating marathon-logger with **loggly**. Marathon Logger generates a rsyslog configuration file so as to pass logs to `syslog` which can be consumed by loggly. But the problem here was that length of mesos's tasks log file path in a configuration file for `rsyslog` cannot be more than `200` characters 😢 . 

We fixed this issue by creating a symlink for the absolute path for mesos logs for tasks running on a particular machine. This made our life easy here for deploying marathon-logger for all our mesos-marathon based systems. 

cc : @ashwanthkumar

### CODE REVIEW

**Comment by aayushKumarJarvis** [comment]
@brewkode, there were a few fixes in the PR which are now done.

File: glide.lock [hunk 1]
```diff
@@ -34,7 +34,7 @@ imports:
   - publicsuffix
 testImports:
 - name: github.com/davecgh/go-spew
-  version: 5215b55f46b2b919f50a1df0eaa5886afe4e3b3d
+  version: 2df174808ee097f90d259e432cc04442cf60be21
```
Comment (ashwanthkumar) [inline glide.lock]: @aayushKumarJarvis if this update is not required as part of this change, please revert this.

File: rsyslog-logger.go [hunk 2]
```diff
@@ -21,18 +22,16 @@ const RsyslogTemplate = `
...
 ######################################
 
-module(load="imfile")
-
 input(type="imfile"
-			File="{{ .CWD }}/{{ .FileName }}"
-			Tag="{{ .CleanAppName }}	{{.TaskID}}"
-			statefile="{{ .TaskID }}"
-      Severity="info")
+		File="{{ .WorkDir }}/{{ .FileName }}"
```
Comment (ashwanthkumar) [inline rsyslog-logger.go]: @aayushKumarJarvis We always seem to have some space vs tab issue in here. Can you please fix this?

File: rsyslog-logger.go [hunk 3]
```diff
@@ -44,13 +43,24 @@ func (r *Rsyslog) AddTask(taskInfo TaskInfo) error {
...
 		return err
 	}
-	// TODO - Support multiple file configurations
-	configFileLocation := fmt.Sprintf("%s/%s-%s.conf", r.ConfigLocation, CommonPrefixToConfigFiles, taskInfo.TaskID)
+
+	// We create symlink of the tasks' sandbox dir so that,
+	// we can reduce the length of the file path that we provide as
+	// part of configs in rsyslog's imfile directive.
+	// When the full file path > 200, rsyslog crashes with "buffer overflow" (evil smile)
+	err = os.Symlink(taskInfo.CWD, fmt.Sprintf("%s/%s", r.WorkDir, taskInfo.TaskID))
```
Comment (ashwanthkumar) [inline rsyslog-logger.go]: @aayushKumarJarvis Can we please extract `fmt.Sprintf("%s/%s", r.WorkDir, taskInfo.TaskID)` into a private function inside `Rsyslog` ? We're using it at another place and it's best if the whole thing is at a single place.

File: task-manager.go [hunk 4]
```diff
@@ -85,42 +86,44 @@ func (t *TaskManager) run() {
...
-					t.KnownTasks[task.TaskID] = time.Now()
-					for _, file := range logFiles {
-						taskInfo := TaskInfo{
-							App:      task.App,
-							Hostname: task.Hostname,
-							Labels:   task.Labels,
-							TaskID:   task.TaskID,
-							CWD:      executor.Directory,
-							FileName: file,
+	//		if task.Hostname == hostname {
```
Comment (ashwanthkumar) [inline task-manager.go]: @aayushKumarJarvis Please revert this section to the one in the master. Check the diff above.

**Review by ashwanthkumar — CHANGES_REQUESTED** [review]

**Comment by ashwanthkumar** [comment]
@aayushKumarJarvis Made the fixes, merging this.

**Pull request merged**